From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Wed, 11 Feb 2026 14:53:55 +0100
Subject: fix(ppcmac): use return instead of exit in pre-udev hook

The `load-thermal.sh` pre-udev hook calls `exit 0`. This causes the
sourcing script (`dracut-pre-udev.sh` on systemd or `init.sh` from the
base Dracut module) to exit. The subsequent hooks are not executed.

Use `return` instead of `exit` in the ppcmac pre-udev hook.

Fixes: https://github.com/dracut-ng/dracut-ng/issues/2227
Forwarded: https://github.com/dracut-ng/dracut-ng/pull/2238
---
 modules.d/70ppcmac/load-thermal.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/70ppcmac/load-thermal.sh b/modules.d/70ppcmac/load-thermal.sh
index 72f2581..0044efd 100755
--- a/modules.d/70ppcmac/load-thermal.sh
+++ b/modules.d/70ppcmac/load-thermal.sh
@@ -3,7 +3,7 @@
 # This hook attempts to load the appropriate thermal modules
 # for PowerPC Macs depending on the specific machine you have.
 
-[ -r /proc/cpuinfo ] || exit 0
+[ -r /proc/cpuinfo ] || return 0
 
 load_windfarm() {
     local pm_model
